lighty-modules/lighty-bgp/src/main/java/io/lighty/modules/bgp/deployer/BgpModule.java

```diff
@@ -0,0 +1,151 @@
+import io.lighty.core.controller.api.LightyServices;
+import io.lighty.modules.bgp.config.InitialBgpConfigLoader;
+import io.netty.channel.EventLoopGroup;
+import java.util.List;
+import org.opendaylight.mdsal.binding.api.DataBroker;
+import org.opendaylight.mdsal.binding.api.RpcProviderService;
+import org.opendaylight.mdsal.binding.dom.codec.api.BindingCodecTree;
+import org.opendaylight.mdsal.binding.dom.codec.api.BindingCodecTreeFactory;
+import org.opendaylight.mdsal.binding.dom.codec.api.BindingNormalizedNodeSerializer;
+import org.opendaylight.mdsal.binding.runtime.api.BindingRuntimeContext;
+import org.opendaylight.mdsal.dom.api.DOMDataBroker;
+import org.opendaylight.mdsal.singleton.common.api.ClusterSingletonServiceProvider;
+import org.opendaylight.protocol.bgp.openconfig.routing.policy.impl.DefaultBGPRibRoutingPolicyFactory;
+import org.opendaylight.protocol.bgp.openconfig.routing.policy.spi.SimpleStatementRegistry;
+import org.opendaylight.protocol.bgp.openconfig.routing.policy.statement.StatementActivator;
+import org.opendaylight.protocol.bgp.openconfig.spi.BGPTableTypeRegistryConsumer;
+import org.opendaylight.protocol.bgp.openconfig.spi.DefaultBGPTableTypeRegistryConsumer;
+import org.opendaylight.protocol.bgp.parser.spi.BGPExtensionConsumerContext;
+import org.opendaylight.protocol.bgp.parser.spi.pojo.DefaultBGPExtensionConsumerContext;
+import org.opendaylight.protocol.bgp.rib.impl.BGPDispatcherImpl;
+import org.opendaylight.protocol.bgp.rib.impl.ConstantCodecsRegistry;
+import org.opendaylight.protocol.bgp.rib.impl.StrictBGPPeerRegistry;
+import org.opendaylight.protocol.bgp.rib.impl.config.DefaultBgpDeployer;
+import org.opendaylight.protocol.bgp.rib.impl.state.BGPStateCollector;
+import org.opendaylight.protocol.bgp.rib.spi.DefaultRIBExtensionConsumerContext;
+import org.opendaylight.protocol.bgp.rib.spi.RIBExtensionConsumerContext;
+import org.opendaylight.protocol.bgp.state.StateProviderImpl;
+import org.opendaylight.yangtools.yang.model.api.EffectiveModelContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class BgpModule extends AbstractLightyModule {
+    private static final Logger LOG = LoggerFactory.getLogger(BgpModule.class);
+    private static final String DEFAULT_BGP_NETWORK_INSTANCE_NAME = "global-bgp";
+
+    private final SimpleStatementRegistry simpleStatementRegistry;
+    private final RIBExtensionConsumerContext ribExtensionConsumerContext;
+    private final BGPDispatcherImpl bgpDispatcher;
+    private final DefaultBGPRibRoutingPolicyFactory routingPolicyFactory;
+    private final ConstantCodecsRegistry codecsRegistry;
+    private final BGPTableTypeRegistryConsumer tableTypeRegistryConsumer;
+    private final DefaultBgpDeployer bgpDeployer;
+    private final BGPStateCollector stateCollector;
+    private final StateProviderImpl stateProvider;
+    private final StrictBGPPeerRegistry peerRegistry;
+    private final InitialBgpConfigLoader initialConfigLoader;
```

Comment:
Attributes: ribExtensionConsumerContext, routingPolicyFactory, codecsRegistry, tableTypeRegistryConsumer, stateCollector could be local variables.

Reply:
done